### PR TITLE
User sign up

### PR DESCRIPTION
**Pull request by akintolga**
- Initial sign up page for new users

### CODE REVIEW

**Comment by akintolga** [comment]
@petrjasek This pr relies on https://github.com/superdesk/superdesk-core/pull/1013
I thought it would be better to keep auth, users and company logic in content-api
Let me know what you think

**Comment by petrjasek** [comment]
I would rather have it here, eventually everything related to content api should go here imo, there won't be any use for it in superdesk

File: newsroom/auth.py [hunk 1]
```diff
@@ -8,15 +14,95 @@
...
+            user = get_resource_service('users').find_one(req=None, _id=user['_id'])
+
+            flask.session['name'] = user.get('name')
             return flask.redirect(flask.url_for('news.index'))
-        kwargs['credentials_error'] = True
-        kwargs['username'] = flask.request.form.get('username', '')
-    return flask.render_template('auth_login.html', **kwargs)
+        except Exception as ex:
+            flask.flash('Invalid login: {} Please try again.'.format(str(ex)))
+            return flask.redirect(flask.url_for('auth.login'))
```
Comment (petrjasek) [inline newsroom/auth.py]: I'm not sure if we should redirect on error in general, imo it's better to render the page again and keep the data where appropriate (like email here).

File: newsroom/templates/auth_login.html [hunk 2]
```diff
@@ -7,8 +7,18 @@
 <p>Invalid credentials.</p>
 {% endif %}
 
+{% with messages = get_flashed_messages() %}
```
Comment (petrjasek) [inline newsroom/templates/auth_login.html]: imo this should go into layout so we don't repeat it

**Comment by petrjasek** [comment]
would be good to use gettext for all strings

**Comment by akintolga** [comment]
gettext is done.

File: newsroom/templates/signup.html [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        <option value="Sales">Sales Representative</option>
+        <option value="CTO">CTO</option>
+        <option value="CEO">CEO</option>
+    </select></p>
+
+    <label><b>Size of your company</b></label>
+    <p><select name="company_size">
+        <option value="0-10">0-10</option>
+        <option value="11-100">11-100</option>
+        <option value=">100">>100</option>
```
Comment (jerome-poisson) [inline newsroom/templates/signup.html]: should not the **>** be escaped to **&amp;gt;**?

File: newsroom/auth.py [hunk 4]
```diff
@@ -1,4 +1,11 @@
 import flask
+from newsroom.utils import query_resource
+from superdesk import get_resource_service
+from superdesk.utc import utcnow
+from datetime import timedelta
+from flask import current_app as app
+import bcrypt
+from flask_babel import gettext
```
Comment (jerome-poisson) [inline newsroom/auth.py]: gettext is usually imported as **_** to make it more readable

**Pull request merged**